src/machine.rs

```diff
@@ -420,6 +420,81 @@ impl<'tcx> PrimitiveLayouts<'tcx> {
     }
 }
 
+/// A thread's CPU affinity mask determines the set of CPUs on which it is eligible to run.
+// the actual representation depends on the target's endianness and pointer width.
+// See CpuAffinityMask::set for details
+#[derive(Clone)]
+pub(crate) struct CpuAffinityMask([u8; Self::MAX_CPUS / 8]);
+
+impl CpuAffinityMask {
+    // this is compatible with the libc `CPU_SETSIZE` constant and corresponds to the number
+    // of CPUs that a `cpu_set_t` can contain.
+    const MAX_CPUS: usize = 1024;
```

Comment:
Somewhere in machine initialization, this constant should be used to throw an error when `Zmiri-num-cpus` is set to a larger value.

Reply:
there is now an assert there, but I also added validation where the miriflags get parsed